From 0d5382ffa585bb1c575aa7f2a6ef2ee2aa167540 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Wed, 17 Jan 2018 10:06:13 +0200
Subject: [PATCH 06/20] lib: save configuration after playback/parse

The configuration is also saved when the libvlc instance is terminated but this doesn't happen often. Indeed, apps using libvlc will generally hold the libvlc instance during all its lifetime and won't be able to clean it up in a lot of case (app killed by OS to free up memory, crashes, reboot, force close...).
---
 lib/media.c        | 2 ++
 lib/media_player.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/lib/media.c b/lib/media.c
index 0cba403598..741aef3821 100644
--- a/lib/media.c
+++ b/lib/media.c
@@ -668,6 +668,8 @@ void libvlc_media_release( libvlc_media_t *p_md )
     input_item_Release( p_md->p_input_item );
 
     libvlc_event_manager_destroy( &p_md->event_manager );
+    if( p_md->is_parsed )
+        config_AutoSaveConfigFile( p_md->p_libvlc_instance->p_libvlc_int );
     libvlc_release( p_md->p_libvlc_instance );
     free( p_md );
 }
diff --git a/lib/media_player.c b/lib/media_player.c
index 5a76610d89..0156ebc9e3 100644
--- a/lib/media_player.c
+++ b/lib/media_player.c
@@ -970,6 +970,8 @@ void libvlc_media_player_pause( libvlc_media_player_t *p_mi )
     vlc_player_TogglePause(player);
 
     vlc_player_Unlock(player);
+
+    config_AutoSaveConfigFile( p_mi );
 }
 
 /**************************************************************************
-- 
2.32.0 (Apple Git-132)

